BinaryObjectScanner/Protection/SecuROM.cs

```diff
@@ -341,5 +342,138 @@ private static string GetV8WhiteLabelVersion(PortableExecutable exe)
+            if (matroschka == null)
+                return null;
+            
+            // Check for all 0x00 required, as at least one known non-RC matroschka has the field, just empty. 
+            if (matroschka.KeyHexString == null || matroschka.KeyHexString == "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0")
+                return "SecuROM Matroschka Package";
+
+            // TODO: I know this reads like debug output, but this should never happen, so I think I'd always want it there?
+            if (matroschka.Entries == null)
+                return "SecuROM Matroschka Package - No Entries? Please report"; 
+                
+            var entry = matroschka.Entries[1];
+            
+            // TODO: I know this reads like debug output, but this should never happen, so I think I'd always want it there?
+            if (entry.MD5 == null)
+                return "SecuROM Matroschka Package - No MD5? Please report"; 
+
+            string md5String = BitConverter.ToString(entry.MD5!);
+            md5String = md5String.ToUpperInvariant().Replace("-", string.Empty);
+            
+            // Check if encrypted executable is known via hash
+            string gameName;
+            var fileData = matroschka.ReadFileData(entry, includeDebug); // Not used yet, but will be in the future
+
+            if (MatroschkaHashDictionary.TryGetValue(md5String, out gameName))
+            {
+                // Returning "SecuROM Matroschka Package" technically redundant since implied.
+                // Since non-debug is more common, return first
+                if (!includeDebug)
+                    return "SecuROM Release Control"; 
+                
+                // TODO: I'd like to have this debug output be the normal output, but I assume I'm not allowed to do that
+                return $"SecuROM Release Control -  {gameName}";
+            }
+            
+            // If not known, check if encrypted executable is likely an alt signing of a known executable
+            // Filetime could be checked here, but if it was signed at a different time, the time will vary anyways
+            var readPathBytes = entry.Path;
+            if (readPathBytes == null)
+                readPathBytes = [];
+
+            string? pathName;
+            if (MatroschkaSizeFilenameDictionary.TryGetValue(entry.Size, out pathName) && pathName == Encoding.ASCII.GetString(readPathBytes).TrimEnd('\0'))
+                return $"SecuROM Release Control - Unknown possible alt executable of size {entry.Size}, please report to us on Github!";
+
+            var readPathName = Encoding.ASCII.GetString(readPathBytes);
+            return $"SecuROM Release Control - Unknown executable {exe.Filename},{readPathName},{md5String},{entry.Size}, PLEASE REPORT ON GITHUB IMMEDIATELY!!!";
+        }
+        
+        // Matches hash of the Release Control-encrypted executable to known hashes
```

Comment:
This should be a proper summary instead of an inline comment. It allows hovering over the method at the invocation site and getting the use instead of having to navigate.

Reply:
Fixed